driver-core/src/main/com/mongodb/connection/CommandMessage.java

```diff
@@ -21,4 +21,16 @@
     CommandMessage(final String collectionName, final OpCode opCode, final MessageSettings settings) {
         super(collectionName, opCode, settings);
     }
+
+    protected static OpCode getOpCode(final MessageSettings settings) {
+        return useNewOpCode(settings) ? OpCode.OP_MSG : OpCode.OP_QUERY;
+    }
+
+    protected static boolean useNewOpCode(final MessageSettings settings) {
```

Comment:
Nit: I'm not a fan of the name - `useNewOpCode` - what if it ages? How about `useOpMsg` ?